src/fe-common/core/fe-windows.c

```diff
@@ -400,71 +440,112 @@ WINDOW_REC *window_find_item(SERVER_REC *server, const char *name)
 
 int window_refnum_prev(int refnum, int wrap)
 {
-	GSList *tmp;
-	int prev, max;
+	GSequenceIter *iter, *end;
 
-	max = prev = -1;
-	for (tmp = windows; tmp != NULL; tmp = tmp->next) {
-		WINDOW_REC *rec = tmp->data;
+	iter = g_sequence_search(windows_seq, GINT_TO_POINTER(refnum-1), (GCompareDataFunc)window_refnum_lookup, NULL);
+	end = g_sequence_get_end_iter(windows_seq);
+
+	if (iter != NULL) {
+		WINDOW_REC *rec;
+		if (iter != end) {
```

Comment:
If I understand this correctly this is a trick to avoid doing the `lookup(refnum)` and `iter_prev` in the common case (is it a common case ?) where you jump to the next/prev window. If so it needs to be documented.